tests/test_utils.py

```diff
@@ -0,0 +1,206 @@
+        assert vocab_handler.vocab_mimetype == 'text/xml'
+        assert '<?xml version="1.0" encoding="UTF-8"?>' in vocab_handler.vocab_file
+
+    def test_py_response(self):
+        vocab = VocabularyFactory()
+        vocab_handler = vocabulary_handler.VocabularyHandler().py_response(vocab)
+        assert vocab_handler.vocab == vocab
+        assert isinstance(vocab_handler, vocabulary_handler.VocabularyHandler)
+
+    def test_create_py(self):
+        vocab_handler = vocabulary_handler.VocabularyHandler().py_response(VocabularyFactory())
+        vocab_handler.create_py()
+        assert vocab_handler.vocab_mimetype == 'text/plain'
+
+    def test_json_response(self):
+        vocab = VocabularyFactory()
+        vocab_handler = vocabulary_handler.VocabularyHandler().json_response(vocab)
+        assert vocab_handler.vocab == vocab
+        assert isinstance(vocab_handler, vocabulary_handler.VocabularyHandler)
+
+    def test_create_json(self):
+        vocab_handler = vocabulary_handler.VocabularyHandler().json_response(VocabularyFactory())
+        vocab_handler.create_json()
+        assert vocab_handler.vocab_mimetype == 'application/json'
+
+    def test_tkl_response(self):
+        vocab = VocabularyFactory()
+        vocab_handler = vocabulary_handler.VocabularyHandler().tkl_response(vocab)
+        assert vocab_handler.vocab == vocab
+        assert isinstance(vocab_handler, vocabulary_handler.VocabularyHandler)
+
+    def test_create_tkl(self):
+        prop = PropertyFactory(property_name='linkback')
+        term = prop.term_key
+        vocab = term.vocab_list
+
+        vocab_handler = vocabulary_handler.VocabularyHandler().tkl_response(vocab)
+
+        root = objectify.fromstring(vocab_handler.vocab_file)
+
+        # Check that the xml doc has all the expected elements, values, and attributes.
+        assert root.tag == 'authority'
+        assert root.get('creator') == vocab.maintainer
+        assert root.get('created') == str(vocab.created).replace(' ', ', ')
+        assert root.get('modifier') == vocab.maintainer
+        assert root.get('modified') == str(vocab.modified).replace(' ', ', ')
+        assert root.enum.get('value') == term.name
+        assert root.enum.get('order') == '1'
+        assert root.enum.string.get('{http://www.w3.org/XML/1998/namespace}lang') == 'en'
+        assert root.enum.string == term.label
+        assert root.enum.linkback == prop.label
+        assert vocab_handler.vocab_mimetype == 'text/xml'
+        assert '<?xml version="1.0" encoding="UTF-8"?>' in vocab_handler.vocab_file
+
+    def test_create_tkl_order_by_name(self):
+        vocab = VocabularyFactory(order='name')
+        TermFactory.create_batch(4, vocab_list=vocab)
+        vocab_handler = vocabulary_handler.VocabularyHandler().tkl_response(vocab)
+
+        root = objectify.fromstring(vocab_handler.vocab_file)
+
+        sorted_terms = vocab.term_set.order_by('name')
+        for i in range(4):
```

Comment:
I'll try iterating over both lists at the same time and see how it goes. Otherwise yeah I'll create a var for that.